app/scripts/views/settings/recovery_codes.js

```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const $ = require('jquery');
+const Cocktail = require('cocktail');
+const FormView = require('../form');
+const ModalSettingsPanelMixin = require('../mixins/modal-settings-panel-mixin');
+const Template = require('templates/settings/recovery_codes.mustache');
+const RecoveryCodePrintTemplate = require('templates/settings/recovery_codes_print.mustache');
+const RecoveryCode = require('../../models/recovery-code');
+
+var t = FormView.t;
+
+const View = FormView.extend({
+  template: Template,
+  className: 'recovery-codes',
+  viewName: 'settings.two-step-authentication.recovery-codes',
+
+  events: {
+    'click .copy-codes': FormView.preventDefaultThen('_copyCodes'),
+    'click .download-codes': '_downloadCodes',
+    'click .print-codes': FormView.preventDefaultThen('_printCodes'),
+    'click .replace-codes-link': FormView.preventDefaultThen('_replaceRecoveryCodes'),
+    'click .two-step-authentication-done': FormView.preventDefaultThen('_returnToTwoStepAuthentication')
+  },
+
+  _returnToTwoStepAuthentication() {
+    this.navigate('settings/two_step_authentication');
+  },
+
+  _copyCodes() {
+    // This copies the recovery codes to clipboard by creating a tiny transparent
+    // textArea with recovery code contents. Then it executes the
+    // browser `copy` command.
+    $('<textArea class=\"recovery-code-text-area\"></textArea>').appendTo('#recovery-codes');
+    this.$('.recovery-code-text-area').html(this.recoveryCodesText);
+    this.$('.recovery-code-text-area').select();
+    this.$('.recovery-code-text-area').focus();
+    try {
+      this.window.document.execCommand('copy');
+      this._displaySuccess(t('Codes copied'));
+    } catch (err) {
+      this._displayError(t('Failed to copy codes'));
+    }
+    this.$('.recovery-code-text-area').remove();
+  },
+
+  _downloadCodes() {
+    // This dynamically creates a link with a blob data of the recovery
+    // codes and clicks it to initiate download.
+    const codeBlob = new Blob([this.recoveryCodesText], {type: 'text/plain'});
+    const filename = 'Recovery Codes.txt';
+    const href = URL.createObjectURL(codeBlob);
+    const template = `
+      <a id="recovery-code-download-link" href="${href}" download="${filename}"></a>
+    `;
+    $(template).appendTo('#recovery-codes');
+    this.window.document.getElementById('recovery-code-download-link').click();
```

Comment:
@vladikoff I could not for the life of me figure out why `this.$('.recovery-code-download-link').click();` would not work but using `this.window.document.getElementById('recovery-code-download-link').click();` does. At a glance, do you see anything that I could be missing?

Reply:
If you're selecting by id you want `#` not `.`:

```js
this.$('#recovery-code-download-link').click();
```

Reply:
Whoops, that was a typo in comment. I was selecting by `#`, which didn't work :/

I thought that might be related to removing the element right after, but the link still does not get triggered if that is commented out.

Reply:
Ah, I got nothing in that case, sorry.

Reply:
I suspect two things are at play:

1. For the browser to trigger a download, a real DOM event must be used.
2. jQuery has its own entire built in event handling system, `this.$(<selector>).click()` synthesizes an event within that system, real events are not used. 

Because of 2, the real DOM event needed to trigger the download does not occur, though the real event is created using `this.window.document.getElementById(selector).click()`